### PR TITLE
Add support for Node 10. Upgrade to node 8.15.0 . Don't start on older Node versions

### PR DESCRIPTION
**Pull request by PieterGit**
- don't start on not supported Node 8 or Node 10 LTS versions
- make exception for Node 10.14.1  from Azure)
- incorporates https://github.com/nightscout/cgm-remote-monitor/pull/4134 and https://github.com/nightscout/cgm-remote-monitor/pull/4129

### CODE REVIEW

**Comment by PieterGit** [comment]
@tynbendad can you test this PR on Azure?
@sulkaharo @jpcunningh can you do a code review and confirm if this PR is ok to merge to dev and to include in 0.11?

File: package.json [hunk 1]
```diff
@@ -48,11 +48,10 @@
     }
   },
   "engines": {
-    "node": "8.12.x",
+    "node": "^10.4.2 || ^8.14.1 || ~8.11.1",
```
[the remark this answers is not in the document]
Comment (PieterGit) [reply package.json]: yes, will fix that soon. thaniks.

**Comment by tynbendad** [comment]
i just tested on azure with a pull of branch 201812_node10 and ns started up/displayed ok - log says:
>Selected node.js version 8.11.1. Use package.json file to choose a different version.

**Comment by PieterGit** [comment]
@tynbendad thanks for testing. Are you sure you set `WEBSITE_NODE_DEFAULT_VERSION` to  `8.11.1` (latest Node version on Azure afaict), see e.g.
https://stackoverflow.com/questions/47388220/azure-app-service-node-version-in-package-json-do-not-work
Is this only a warning, or does it not start on Azure?

**Comment by tynbendad** [comment]
yes, it is working fine - i have WEBSITE_NODE_DEFAULT_VERSION set to 8.11.1 under "Application settings".
fwiw it seems to run out of the free quota on azure every day, so its not very useful but i'll keep it around for testing, should work for folks that want to pay for azure.

**Comment by tynbendad** [comment]
it is running out of CPU quota, which appears to be 60 minutes a day. i don't actually look at the azure site so i'm not sure how its using that much CPU.

File: package.json (hunk 1 above)
[the remark this answers is not in the document]
Comment (PieterGit) [reply package.json]: fixed

**Comment by PieterGit** [comment]
@balshor  @tynbendad  can you confirm that this PR or the `needs_testing` branch works with Azure and Node 10.14.1 on Azure? According to  @balshor Microsoft upgraded their Node to 10.14.1 (latest available is Node 10.15.0). See https://gitter.im/nightscout/public?at=5c2a50c70b7fc97caaca6211

Make sure to set `WEBSITE_NODE_DEFAULT_VERSION` to `10.14.1`

@tynbendad can you clarify the CPU quota you mention on https://github.com/nightscout/cgm-remote-monitor/pull/4155#issuecomment-450517190 . Do you mean you can only use Nightscout 0.11-dev on Azure for ~ 1 hour a day on a free Azure tier?

File: lib/server/bootevent.js [hunk 2]
```diff
@@ -6,6 +6,34 @@ var UPDATE_THROTTLE = 1000;
 
 function boot (env, language) {
 
+  //////////////////////////////////////////////////
+  // Check Node version. Latest Node 8 LTS and Latest Node 10 LTS are recommended and supported.
+  // < 8        does not work, not supported
+  // >= 8.14.0  works, supported and recommended
```
Comment (jpcunningh) [inline lib/server/bootevent.js]: I believe this comment should now say `>= 8.15.0`

**Comment by balshor** [comment]
I can confirm that the `needs_testing` branch starts successfully on Azure with Node 10.14.1 (verified via "Selected node.js version 10.14.1" in the deployment logs).

File: lib/server/bootevent.js (hunk 2 above)
Comment (PieterGit) [reply lib/server/bootevent.js]: fixed

**Comment by PieterGit** [comment]
This PR has been tested as part of the `need_testing` branch of https://github.com/nightscout/cgm-remote-monitor/pull/4169
@blocklist_twitter @balshor @sulkaharo @jpcunningh @unsoluble @veryfancy @balshor @tynbendad have run this PR for a week.
No negative stuff popped up and the checks work (even on Azure). Gitter link: https://gitter.im/nightscout/public?at=5c2f33d12863d8612bd78eb0

If you encounter bugs with this PR, please create a new issue and link it to this PR.
Closing PR, will be merged with https://github.com/nightscout/cgm-remote-monitor/pull/4169

**Pull request closed without merging**